### DISCUSSION TITLE
Checkbox is not remove checked box value

### DISCUSSION THREAD

**Discussion by mr-chetan** [livewire/livewire#2990] [7. Bugs (archived)]
The checkbox doesn't remove the checkbox value when it unchecked even it not get checked by the checked attribute.

### In View
```php
<div class="">
    <h3 class="mb-2 text-gray-900 dark:text-gray-100 text-xl">Roles Permissions ID - @json($rolePermissions)</h3>
    <h3 class="mb-2 text-gray-900 dark:text-gray-100 text-xl">Selected ID - @json($selectedPermissions)</h3>
    <div class="bg-white dark:bg-primary-900 rounded-md shadow mt-2 py-3 px-2">
        <h3 class="mb-2 text-gray-900 dark:text-gray-100 text-xl">Permissions</h3>
        <div class="grid grid-cols-12 gap-4">
            @foreach ($permissions as $permission)
                <div class="col-span-6 sm:col-span-4 md:col-span-3 flex items-center space-x-2">
                    <input wire:model="selectedPermissions"
                        {{ in_array($permission->id, $rolePermissions) ? 'checked' : '' }} type="checkbox"
                        value="{{ $permission->id }}"
                        class="w-4 h-4 transition text-primary duration-300 rounded focus:ring-2 focus:ring-offset-0 focus:outline-none focus:ring-primary-200 dark:focus:ring-primary-700">
                    <label class="text-sm font-semibold text-gray-500 dark:text-gray-100">
                        {{ ucwords($permission->display_name) }}
                    </label>

                </div>
            @endforeach
        </div>
    </div>
</div>
```

### In Livewire Component Class
```php
 public $role;
    public $permissions;
    public $selectedPermissions = [];
    public $rolePermissions = [];
    protected $listeners = ['eidtRole'];

    public function mount()
    {
        $this->permissions = Permission::get();
    }
    public function eidtRole(Role $role)
    {
        $this->role = $role;
        $this->rolePermissions = $role->permissions() ->pluck('id')->toArray();
        $this->selectedPermissions=$this->rolePermissions;
    }
```
https://user-images.githubusercontent.com/56998650/122450316-81041480-cfc4-11eb-8d50-2727d62fdc07.mp4

When I remove `$this->selectedPermissions=$this->rolePermissions;` from Class checkbox is not check

in input, I already add condition ` {{in_array($permission->id, $rolePermissions)?"checked":""}}` but id didn't check the checkbox 

https://user-images.githubusercontent.com/56998650/122450604-d2ac9f00-cfc4-11eb-95ec-fa96651bd114.mp4

**Context**
- Livewire version: [2.5.1]
- Laravel version: [8.47.0]
- Alpine version: [3.0.6]
- Browser: [Chrome]

**Comment by joshhanley** [comment]
@mr-chetan Livewire has issues with checkboxes and numeric id's in an array. 

The issue is checkboxes submit as strings, but your array is integers and isn't seen as the same thing (hence why in your video you see some permissions are integers and others strings)

![image](https://user-images.githubusercontent.com/882837/122476793-18d82180-d00a-11eb-89d4-9253f9bd6373.png)

The quick fix is to convert all integers in your $rolePermissions array to strings like

```php
$this->rolePermissions = $role->permissions() ->pluck('id')->map(function($id) {
    return "{$id}";
})->toArray();
```

Hope this helps!

**Comment by mr-chetan** [comment]
Yes, it is work but why I need to convert every value in string even i apply it in numeric value. 

Livewire use string value instead of numeric.

I think this should be fixed.

I use this way to fix it now 

```php
$this->rolePermissions = $role->permissions()->pluck('id')->map(function ($id) {
      return  strval($id);
})->toArray();
```

**Comment by chillbram** [comment]
This is indeed still a bug: see https://github.com/livewire/livewire/issues/788
